sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/ModelsRepositoryAsyncClient.java

```diff
@@ -62,4 +63,64 @@
     public ModelsRepositoryServiceVersion getServiceVersion() {
         return this.serviceVersion;
     }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmi A well-formed DTDL model Id. See <a href="https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md">DTDL specs</a>. For example 'dtmi:com:example:Thermostat;1'.
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Map<String, String>> getModels(String dtmi) {
+        return getModels(dtmi, this.defaultDependencyResolutionOption);
+    }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmi                 A well-formed DTDL model Id. See <a href="https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md">DTDL specs</a>. For example 'dtmi:com:example:Thermostat;1'.
+     * @param dependencyResolution A {@link ModelsDependencyResolution} value to force model resolution behavior.
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Map<String, String>> getModels(String dtmi, ModelsDependencyResolution dependencyResolution) {
+        return withContext(context -> getModels(dtmi, dependencyResolution, context));
+    }
+
+    Mono<Map<String, String>> getModels(String dtmi, ModelsDependencyResolution dependencyResolution, Context context) {
+        context.addData(AZ_TRACING_NAMESPACE_KEY, MODELS_REPOSITORY_TRACING_NAMESPACE_VALUE);
+        return repositoryHandler.processAsync(dtmi, dependencyResolution, context);
+    }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmis Collection of well-formed DTDL model Ids
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
```

Comment:
Good to add what's the default behavior of dependency resolution in comments

Reply:
We use the parameter that they use in the Client builder so we can't definitively say which behavior we will be using.

Reply:
Gotcha